GUI/AllegroInput.cpp

```diff
@@ -239,13 +239,63 @@ void AllegroInput::Update(void)
 	}
 	else 
 	{
-		m_MouseButtonsEvents[0] = mouse_b & AMBLEFT ? (m_MouseButtonsStates[0] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[0] == Down ? Released : None);
-		m_MouseButtonsEvents[1] = mouse_b & AMBMIDDLE ? (m_MouseButtonsStates[1] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[1] == Down ? Released : None);
-		m_MouseButtonsEvents[2] = mouse_b & AMBRIGHT ? (m_MouseButtonsStates[2] == Up ? Pushed : Repeat) : (m_MouseButtonsStates[2] == Down ? Released : None);
-
-		m_MouseButtonsStates[0] = mouse_b & AMBLEFT ? Down : Up;
-		m_MouseButtonsStates[1] = mouse_b & AMBMIDDLE ? Down : Up;
-		m_MouseButtonsStates[2] = mouse_b & AMBRIGHT ? Down : Up;
+		// Left mouse button
+		if (mouse_b & AMBLEFT) {
+			if (m_MouseButtonsStates[0] == Up) { // Previously was up
```

Comment:
I think you can compact these into a ternary. Nested ternaries like what was there are pretty unreadable, but one simple one inside an if is fine. i.e.
```
if (mouse_b & AMBLEFT) {
	m_MouseButtonEvents[0] = (m_MouseButtonStates[0] == Up) ? Pushed : Repeat;
}
```
This applies to all of these similar ones. Also, you should clean up the kinda random comments.